src/log.c

```diff
@@ -520,6 +524,7 @@ void qd_log_finalize(void) {
         qd_log_entry_free_lh(DEQ_HEAD(entries));
     while (DEQ_HEAD(sink_list))
         log_sink_free_lh(DEQ_HEAD(sink_list));
+    default_log_source = NULL;  // stale value would end up misconfiguring router again in the same process
```

Comment:
Global variable that needs to be NULLed out on exit, otherwise starting again in the same process segfaults on logging.